app/controllers/locations_controller.rb

```diff
@@ -14,23 +14,23 @@ def external_search
     results = []
     query = location_params[:query]
     limit = location_params[:limit]
+
     if query.present?
-      success, resp = Location.geosearch(query, limit)
-      if success && resp.is_a?(Array)
-        # Successful request with results. Just keep the first if you get duplicate locations.
-        results = resp.map { |r| LocationHelper.adapt_location_iq_response(r) }
-        results = results.uniq { |r| [r[:name], r[:geo_level]] }
-      elsif success && resp.is_a?(Hash) && resp["error"] == "Unable to geocode"
-        # Successful request but 0 results
-        Rails.logger.info("No geosearch results for: #{query}")
-      else
-        # Unsuccessful request. Likely Net::HTTPTooManyRequests. Monitor if users run up against geosearch API rate limits / record any other errs.
-        msg = GEOSEARCH_RATE_LIMIT_ERR
-        msg += ": #{resp}" if resp
-        LogUtil.log_err_and_airbrake(msg)
-        raise msg
+      responses = {}
+      threads = []
+      Location::GEOSEARCH_ACTIONS.each do |action|
+        t = Thread.new { external_search_action(action, query, limit, responses) }
+        # Suppress verbose backtrace on handled exceptions
+        t.report_on_exception = false
```

Comment:
Nit: Should we suppress this? Since you don't have a `rescue` block around the entire `external_search_action`, if an unexpected error happens, I think it will fail silently?

Would it be better to not suppress this, and also not re-raise exceptions that are expected (such as the rate limit one) if we are logging them?

I notice this is the first time we're using multi-threading in the code base, so paying a bit more attention to conventions.

Reply:
Actually the errors inside will still be raised when `join` is called (and the unit tests include this). The difference is that if this is `true` it would exit immediately and you will see a verbose stdout when running tests even if it was expected and handled. Will clarify this code comment.

`If it is guaranteed the Thread will be joined with #join or #value, then it is safe to disable this report with Thread.current.report_on_exception = false when starting the Thread` https://ruby-doc.org/core-2.5.0/Thread.html#method-c-report_on_exception

It would look like this when unsuppressed but successful ;)
<img width="931" alt="Screen Shot 2019-10-16 at 1 55 42 PM" src="https://user-images.githubusercontent.com/5652739/66958265-ab961100-f01c-11e9-85ce-f083f3b18681.png">

Reply:
Ah I see. So `report_on_exception` will do an additional log.

Reply:
Thanks for clarifying!